### PR TITLE
8324326: Update ICU4C to 74.2

### PR DESCRIPTION
**Pull request by HimaBinduMeda**
Updated icu to v74.2. Sanity testing looks fine. Verified build on all platforms

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (2 reviews required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer), 1 [Author](https://openjdk.org/bylaws#author))

### Issue
 * [JDK-8324326](https://bugs.openjdk.org/browse/JDK-8324326): Update ICU4C to 74.2 (**Bug** - P3)

### Reviewers
 * [Kevin Rushforth](https://openjdk.org/census#kcr) (@kevinrushforth - **Reviewer**)
 * [Joeri Sykora](https://openjdk.org/census#sykora) (@tiainen - Author)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jfx.git pull/1398/head:pull/1398` \
`$ git checkout pull/1398`

Update a local copy of the PR: \
`$ git checkout pull/1398` \
`$ git pull https://git.openjdk.org/jfx.git pull/1398/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1398`

View PR using the GUI difftool: \
`$ git pr show -t 1398`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jfx/pull/1398.diff">https://git.openjdk.org/jfx/pull/1398.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jfx/pull/1398#issuecomment-1991942083)

### CODE REVIEW

File: modules/javafx.web/src/main/native/Source/ThirdParty/icu/CMakeLists.txt [hunk 1]
```diff
@@ -588,7 +590,7 @@ target_link_libraries(data_as_asm icuuc icui18n)
 file(GLOB ICU_DATA_ZIP_FILE "${CMAKE_BINARY_DIR}/icu/data/icu4c-*.zip")
 
 # Get just file name without extension
-set(ICU_DATA_FILE_NAME "icudt73l")
+set(ICU_DATA_FILE_NAME "icudt74l")
```
Comment (kevinrushforth) [inline modules/javafx.web/src/main/native/Source/ThirdParty/icu/CMakeLists.txt:593]: Shouldn't this be 742?

**Comment by kevinrushforth** [comment]
@johanvos or @tiainen can one of you be the second reviewer?

**Review by kevinrushforth — COMMENTED** [review]
I haven't looked at it detail or tested it yet, but I did spot one thing that raises a question.

File: modules/javafx.web/src/main/native/Source/ThirdParty/icu/CMakeLists.txt (hunk 1 above)
Comment (kevinrushforth) [reply modules/javafx.web/src/main/native/Source/ThirdParty/icu/CMakeLists.txt:593]: Never mind. I see now that's an `l` not a `1` after the `74. Sorry for the noise.

Comment (HimaBinduMeda) [reply modules/javafx.web/src/main/native/Source/ThirdParty/icu/CMakeLists.txt:593]: icu4c-74_2-data-bin-l contains the dat file named as icudt74l.dat. Hence, this is correct.

**Review by tiainen — APPROVED** [review]
Builds and tests all ran fine.

**Review by kevinrushforth — APPROVED** [review]
Looks good.

**Pull request closed without merging**